### DISCUSSION TITLE
v1.6.0

### DISCUSSION THREAD

**Discussion by edespino** [apache/cloudberry#616] [Announcements]
## 📝️ What's Changed (Pull Request list)
* Enhance: Add warehouse_id to pg_stat_activity_extended by @lss602726449 in https://github.com/cloudberrydb/cloudberrydb/pull/453
* Modify the orca optimizer's processing of unionall distribution strat… by @Light-City in https://github.com/cloudberrydb/cloudberrydb/pull/399
* Update the location of python-dependencies.txt by @Zhangbaowen-Hashdata in https://github.com/cloudberrydb/cloudberrydb/pull/460
* Add hook for exec_simple_query and support self-defined spilling memory threshold in cost module. by @songdongxiaoa2 in https://github.com/cloudberrydb/cloudberrydb/pull/447
* Fix compile error that pipe is redefined by @gfphoenix78 in https://github.com/cloudberrydb/cloudberrydb/pull/349
* [AQUMV] Use contain_var_clause instead of pull_var_clause. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/451
* directory_table: matchignore & remove meta-command username output by @edespino in https://github.com/cloudberrydb/cloudberrydb/pull/464
* Enhance: Add hook to support different way to create/drop/alter wareh… by @lss602726449 in https://github.com/cloudberrydb/cloudberrydb/pull/462
* Fix: replace gpscp with gpsync in ext fts by @jiaqizho in https://github.com/cloudberrydb/cloudberrydb/pull/470
* Re-used index after ALTER COLUMN TYPE shouldn't change relfilenode by @lss602726449 in https://github.com/cloudberrydb/cloudberrydb/pull/474
* Cherry-pick Resgroup related code from GreenPlum [Mar 2, 2022 - Feb 7, 2023] by @foreyes in https://github.com/cloudberrydb/cloudberrydb/pull/448
* Add weekly build and release process by @Zhangbaowen-Hashdata in https://github.com/cloudberrydb/cloudberrydb/pull/459
* [Answer Query Using Materialized Views] Consider mutable functions and bypass expressions have no Vars for view query target.  by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/455
* [AQUMV] Expose the function adjust view query and varno fix. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/469
* Modify weekly build release details by @Zhangbaowen-Hashdata in https://github.com/cloudberrydb/cloudberrydb/pull/477
* Call the query_info_collect_hook directly if expection happens by @foreyes in https://github.com/cloudberrydb/cloudberrydb/pull/481
* Fix record password_history when the role is not allowed to use profile. by @wenchaozhang-123 in https://github.com/cloudberrydb/cloudberrydb/pull/480
* Resolve inconsistent result issue in gpdtm_plpgsql test case by @congxuebin in https://github.com/cloudberrydb/cloudberrydb/pull/491
* Fix cbdb ci not running properly by @Zhangbaowen-Hashdata in https://github.com/cloudberrydb/cloudberrydb/pull/497
* cherry-pick: Work around spurious compiler warning in inet operators by @gfphoenix78 in https://github.com/cloudberrydb/cloudberrydb/pull/499
* Add LoongArch(loongarch64) support to s_lock.h by @wangzw in https://github.com/cloudberrydb/cloudberrydb/pull/500
* Implement some feature of directory table. by @wenchaozhang-123 in https://github.com/cloudberrydb/cloudberrydb/pull/484
* Fix copy to replicated table display copy number is incorrect. by @wenchaozhang-123 in https://github.com/cloudberrydb/cloudberrydb/pull/498
* Fix the correct size of memory block in bitmapinsert by @gfphoenix78 in https://github.com/cloudberrydb/cloudberrydb/pull/495
* re-enable the external FTS ICW by @jiaqizho in https://github.com/cloudberrydb/cloudberrydb/pull/483
* Fix ignore of direct table test files. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/502
* Fix gpinitsystem by @fanfuxiaoran in https://github.com/cloudberrydb/cloudberrydb/pull/490
* Fix compile errors detected by gcc 12 by @gfphoenix78 in https://github.com/cloudberrydb/cloudberrydb/pull/503
* Fix bsearch compare function in guc.c by @gfphoenix78 in https://github.com/cloudberrydb/cloudberrydb/pull/507
* [Unstable cases]Fix Some commands forgot to mark meta track. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/505
* Fix compile error in c++20 by @gfphoenix78 in https://github.com/cloudberrydb/cloudberrydb/pull/510
* change AOCO_Compression test case to validate pg_relation_size and get_ao_compression_ratio within ±10% expected range  by @congxuebin in https://github.com/cloudberrydb/cloudberrydb/pull/493
* Maintain materialized view data status. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/501
* Define EXT_OID_START to suggest oid range for extensions. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/514
* Ignore pg_upgrade to make CI happy. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/515
* Reduce flakiness in test fts_segment_reset by @jiaqizho in https://github.com/cloudberrydb/cloudberrydb/pull/518
* make gp_dqa test case stable by @congxuebin in https://github.com/cloudberrydb/cloudberrydb/pull/521
* Doc: list more tools for Cloudberry Database by @tuhaihe in https://github.com/cloudberrydb/cloudberrydb/pull/523
* Fix copy to directory table always return 1. by @wenchaozhang-123 in https://github.com/cloudberrydb/cloudberrydb/pull/522
* Reimplement copy from directory table. by @wenchaozhang-123 in https://github.com/cloudberrydb/cloudberrydb/pull/527
* Add materialized view relative triggers case in SingleNode mode. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/517
* [Answer Query Using Materialized Views] Refactor view query target list process. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/525
* Implementation of Tag. by @wenchaozhang-123 in https://github.com/cloudberrydb/cloudberrydb/pull/444
* Fix segfilecount of AO/AOCO when bulk insertion: COPY by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/530
* Feature: Update orafce to 4.9 and enhance it by @foreyes in https://github.com/cloudberrydb/cloudberrydb/pull/524
* [AQUMV]Allow to use normal materialized views to answer query. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/528
* Support copy from directory table for entry distribution policy. by @wenchaozhang-123 in https://github.com/cloudberrydb/cloudberrydb/pull/533
* Small README.md improvement. by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/534
* Use FDW to query multiple servers as shards by @oppenheimer01 in https://github.com/cloudberrydb/cloudberrydb/pull/320
* Add an option in psql to avoid encoding issues on some platforms by @gfphoenix78 in https://github.com/cloudberrydb/cloudberrydb/pull/535
* Refactor cbdb_log to use vfprintf by @ruhuang2001 in https://github.com/cloudberrydb/cloudberrydb/pull/506
* Update aocsam.c: fix safeFSWriteSize arg type by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/540
* Update the image to Rockylinux8 to ensure the normal operation of CI by @Zhangbaowen-Hashdata in https://github.com/cloudberrydb/cloudberrydb/pull/556
* Remove unsupported AC_FUNC_FSEEKO macro by @gfphoenix78 in https://github.com/cloudberrydb/cloudberrydb/pull/543
* adjust test cases for the cloudberrydb binary swap test by @congxuebin in https://github.com/cloudberrydb/cloudberrydb/pull/537
* CREATE FOREIGN TABLE LIKE. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/554
* Enable SingleQE join with SegmentGeneralWorkers by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/327
* Use syscache lookup in ShouldUseReservedSegno by @gongxun0928 in https://github.com/cloudberrydb/cloudberrydb/pull/541
* Implement READ_STRING_FIELD_NULL serializable read function. by @wenchaozhang-123 in https://github.com/cloudberrydb/cloudberrydb/pull/553
* Fix: free tupleDesc on commit or abort transcation. by @yjhjstz in https://github.com/cloudberrydb/cloudberrydb/pull/551
* Update appendonlywriter.c: fix debug message by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/564
* Fix crash of COPY FROM on AO/AOCO/PAX partition tables. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/549
* Fix: Add ao_unique_index_build test in greenplum_schedule by @lss602726449 in https://github.com/cloudberrydb/cloudberrydb/pull/562
* Fix copy some directory table to problems. by @wenchaozhang-123 in https://github.com/cloudberrydb/cloudberrydb/pull/550
* Fix bug with base relation truncate for IMMV by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/570
* Suppress compile-time error in SparseData.h by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/566
* Support lock directory table. by @wenchaozhang-123 in https://github.com/cloudberrydb/cloudberrydb/pull/572
* Doc: update the alter table help command content by @EcaleD in https://github.com/cloudberrydb/cloudberrydb/pull/574
* Cherry-pick Resgroup V2 and toolkit from GPDB by @jiaqizho in https://github.com/cloudberrydb/cloudberrydb/pull/531
* Fix: no need to do the qual-clause twice by @jiaqizho in https://github.com/cloudberrydb/cloudberrydb/pull/396
* [ORCA]: fallback to postgres optimizer when NL inner plan is index scan by @gfphoenix78 in https://github.com/cloudberrydb/cloudberrydb/pull/565
* Add FIXME about redundant param from 'ivm_visible_in_prestate' call. by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/578
* Support distributed by clause specification when Create MV if not exists by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/563
* Cherry pick commits related to recent RESGROUP v2 merge.  by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/579
* Change temp table name used to IMMV to some less frequent sequence of… by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/581
* Rephrase comment for deferred IVM case by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/576
* Fix pxf_fragment.c compilation failure by @Terry1504 in https://github.com/cloudberrydb/cloudberrydb/pull/590
* Fix pg_upgrade version parsing when upgrading from Greenplum by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/599
* Enable fsync=on by default in gpAux/gpdemo/gpdemo-defaults.sh by @yjhjstz in https://github.com/cloudberrydb/cloudberrydb/pull/585
* Fix serialization of expr AEXPR_NOT_DISTINCT. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/598
* Forbid inherits tables to store into gp_matview_aux. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/587
* [AQUMV] Check relation actually has children when view matching. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/577
* Update check.c: fixed Cloudberry incorrect references by @reshke in https://github.com/cloudberrydb/cloudberrydb/pull/600
* Fix writable rules on table has relative matview. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/584
* Fix writable CTE cause wrong matview data status. by @avamingli in https://github.com/cloudberrydb/cloudberrydb/pull/602
* Send rows in binary mode for ANALYZE by @Light-City in https://github.com/cloudberrydb/cloudberrydb/pull/601
* Enable the resgroup test cases in CI by @jiaqizho in https://github.com/cloudberrydb/cloudberrydb/pull/539
* Fix: could not pull up equivalence class using projected target list by @yjhjstz in https://github.com/cloudberrydb/cloudberrydb/pull/606
* Remove cbdb weekly and release workflows. by @edespino in https://github.com/cloudberrydb/cloudberrydb/pull/615

**Full Changelog:** https://github.com/cloudberrydb/cloudberrydb/compare/1.5.4...1.6.0

---

## 🙌🏻️ New Contributors
* @Light-City made their first contribution in https://github.com/cloudberrydb/cloudberrydb/pull/399
* @songdongxiaoa2 made their first contribution in https://github.com/cloudberrydb/cloudberrydb/pull/447
* @edespino made their first contribution in https://github.com/cloudberrydb/cloudberrydb/pull/464
* @congxuebin made their first contribution in https://github.com/cloudberrydb/cloudberrydb/pull/491
* @wangzw made their first contribution in https://github.com/cloudberrydb/cloudberrydb/pull/500
* @reshke made their first contribution in https://github.com/cloudberrydb/cloudberrydb/pull/534
* @oppenheimer01 made their first contribution in https://github.com/cloudberrydb/cloudberrydb/pull/320
* @ruhuang2001 made their first contribution in https://github.com/cloudberrydb/cloudberrydb/pull/506
* @EcaleD made their first contribution in https://github.com/cloudberrydb/cloudberrydb/pull/574
* @Terry1504 made their first contribution in https://github.com/cloudberrydb/cloudberrydb/pull/590

---

## 🧑🏻‍💻 Contributors

👍  Thanks to all the contributors to make this release happen:
@Aegeaner, @EcaleD, @Light-City, @RMTT, @SmartKeyerror, @Tao-T, @Terry1504, @Zhangbaowen-Hashdata, @adam8157, @airfan1994, @andr-sokolov, @ashwinstar, @avamingli, @beeender, @bmdoil, @charliettxx, @congxuebin, @dgkimura, @dh-cloud, @divyeshddv, @dreamedcheng, @edespino, @eespino, @fairyfar, @fanfuxiaoran, @foreyes, @gfphoenix78, @gongxun0928, @gpopt, @higuoxing, @huansong, @hyongtao-db, @jchampio, @jiaqizho, @jimmyyih, @kainwen, @l-wang, @lss602726449, @oppenheimer01, @reshke, @ruhuang2001, @songdongxiaoa2, @soumyadeep2007, @thedanhoffman, @tuhaihe, @wangzw, @wenchaozhang-123, @yanwr1, @yaowangm, @yjhjstz, @zhrt123, @zxuejing

---

<sub>🧂🧪 Try out Cloudberry Database via building [one Docker-based Sandbox](https://github.com/cloudberrydb/bootcamp), which is tailored to help you gain a basic understanding of Cloudberry Database's capabilities and features a range of materials, including tutorials, sample code, and crash courses.</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/cloudberrydb/cloudberrydb/releases/tag/1.6.0'>v1.6.0</a>.</em>

**Comment by tuhaihe** [comment]
Great work! Thanks @edespino! 👍

**Comment by jianlirong** [comment]
Great work!

**Comment by reshke** [comment]
Great 💯 💯 💯
